src/InsertionsClient.Core/Api/Providers/InsertionApi.cs

```diff
@@ -229,6 +231,13 @@ internal bool TryExtractManifestAssets(string manifestFile, ICollection<Asset> a
 
                 foreach (Build build in buildManifest.Builds)
                 {
+                    if(buildFilter != null && !buildFilter(build))
+                    {
+                        Trace.WriteLine($"The build with {nameof(build.BuildNumber)} {build.BuildNumber} was ignored by the filter. " +
+                            "Assets from this build will not be used in the insertion process.");
+                        continue;
+                    }
+
                     if (build.Assets == null)
                     {
                         Trace.WriteLine("Manifest file contains build with no assets inside. " +
```

Comment:
Should this message be adapted in case manifest file contains the builds but all of them were filtered out?